src/plugins/discover/public/application/main/utils/resolve_data_view.ts

```diff
@@ -122,12 +120,14 @@ export async function loadDataView(
   } catch (e) {}
 
   // fetch persisted data view
-  const actualId = getDataViewId(fetchId, dataViewList, config.get('defaultIndex'));
   return {
```

Comment:
We don't need `getDataViewId` because `dataViews.getDefaultDataView` does most of the work - it provides the default data view and if it doesn't exist then it sets another default and if there aren't any data views...well, thats what the `hasData` elsewhere is for.